File: ManejoArchivosJava/src/ManejoArchivosJava/Barco.java
/*
 *esta es la clase Barco que hereda de la clase vehiculo y ademas tambien 
posee los metodos de la clase cualidades
 */
package ManejoArchivosJava;

import ManejoArchivosJava.Vehiculo;
import javax.swing.JOptionPane;

/**
 * 28 de septiembre del 2018
 * @author Manuela García Monsalve 
 */
public class Barco extends Vehiculo implements Cualidades{
    
    String medio;
    
    
    //constructor de la clase Barco
    public Barco(String marca, String modelo, String color, String medio)
    {
        super(marca,modelo,color);
        this.medio = medio;
    }
    
    //sobrescribir los metodos de la clase barco
    @Override
    public void Prender()
    {
        JOptionPane.showMessageDialog(null, "me he prendido ");
    }
    @Override
    public void Arrancar()
    {
        JOptionPane.showMessageDialog(null, "he arrancado ");
    }
    @Override
    public void Apagar()
    {
        JOptionPane.showMessageDialog(null, "me he apagado ");
    }
    
    @Override
    public void Reversa()
    {
        JOptionPane.showMessageDialog(null, "no Puede dar Reversa");
    }
    
        @Override
    public String VerDatos()
    {
        return  "Marca: " + this.marca 
            + "\nModelo: " + this.modelo + "\nColor: " + this.color
            + "\nMedio: " + this.medio;
    }
}

File: ManejoArchivos/Barco.py
#Autor Manuela Garcia Monsalve
# 28 septiembre 2018
#Esta es la subclase barco que heredara los atributos de la superclase vehiculo

from Vehiculo import *

#La clase barco heredara los atributos de marca,color y modelo de la superclase vehiculo
class Barco(Vehiculo):

    def __init__(self,marca,color,modelo,medio):#Se genera la heriencia de los atributos
        Vehiculo.__init__(marca,color,modelo)
        self.medio = medio

    def Prender(self): #Metodos a sobreescribir
        pass

    def Arrancar(self):
        pass

    def Apagar(self):
        pass